cookbook/entities/datagrid.rst

```diff
@@ -0,0 +1,201 @@
+
+.. _cookbook-entities-grid-config:
+
+Configure the Grid
+------------------
+
+The datagrid configuration happens in the ``datagrid.yml`` file in the configuration directory of
+your bundle and is divided into several sections:
+
+Data Source
+~~~~~~~~~~~
+
+The ``source`` option is used to configure a Doctrine query builder that is used to fetch the data
+to be displayed in the grid:
+
+.. code-block:: yaml
+    :linenos:
+
+    # src/AppBundle/Resources/config/datagrid.yml
+    datagrid:
+        app-tasks-grid:
+            source:
+                type: orm
+                query:
+                    select:
+                        task.id
+                        task.subject
+                        task.description
+                        task.dueDate
+                        priority.label AS taskPriority
+                    from:
+                        - { table: AppBundle:Task, alias: task }
+                    join:
+                        left:
+                            - { join: task.priority, alias: priority }
+
+Displayed Columns
+~~~~~~~~~~~~~~~~~
+
+Then, the ``columns`` option needs to be used to configure how which data will be displayed:
+
+.. code-block:: yaml
+    :linenos:
+
+    # src/AppBundle/Resources/config/datagrid.yml
+    datagrid:
+        app-tasks-grid:
+            # ...
+            columns:
+                id:
+                    label: ID
+                    frontend_type: integer
+                subject:
+                    label: Subject
+                description:
+                    label: Description
+                dueDate:
+                    label: Due Date
+                    frontend_type: datetime
+                taskPriority:
+                    label: Priority
+
+For each column, the ``frontend_type`` option can be used to customize how data will be displayed
+(by default, the data will be shown as is).
+
+Column Sorters
+~~~~~~~~~~~~~~
+
+Use the ``sorters`` option to define on which columns' header the user can click to order by the
+data:
+
+.. code-block:: yaml
+    :linenos:
+
+    # src/AppBundle/Resources/config/datagrid.yml
+    datagrid:
+        app-tasks-grid:
+            # ...
+            sorters:
+                columns:
+                    id:
+                        data_name: task.id
+                    subject:
+                        data_name: task.subject
+                    description:
+                        data_name: task.description
+                    dueDate:
+                        data_name: task.dueDate
+                    taskPriority:
+                        data_name: priority.label
+                default:
+                    dueDate: DESC
+
+Each key under ``sorters.columns`` refers to one of the displayed columns. The ``data_name`` option
+is the term that will be used as the ``order by`` term in the Doctrine query.
+
+Data Filters
+~~~~~~~~~~~~
+
```

Comment:
Please add short description about what filters are and some description about config tree structure like you did for another blocks.